File: core-tests/e2e-tests/spring/spring-rest-h2-v1/src/test/java/com/foo/rest/examples/spring/db/existingdata/ExistingDataController.java
package com.foo.rest.examples.spring.db.existingdata;

import com.foo.rest.examples.spring.db.SpringWithDbController;
import org.evomaster.client.java.sql.DbSpecification;

import java.util.Collections;
import java.util.List;

/**
 * Created by arcuri82 on 19-Jun-19.
 */
public class ExistingDataController extends SpringWithDbController {

    public ExistingDataController() {
        super(ExistingDataApp.class);
    }

    @Override
    public void resetStateOfSUT() {
        super.resetStateOfSUT();

//        ExistingDataEntityX x = new ExistingDataEntityX();
//        x.setId(42L);
//        x.setName("Foo");
//
//        ExistingDataRepositoryX rep = ctx.getBean(ExistingDataRepositoryX.class);
//        rep.save(x);
    }

    @Override
    public List<DbSpecification> getDbSpecifications() {
        List<DbSpecification> spec =  super.getDbSpecifications();
        if (spec != null && !spec.isEmpty())
            return Collections.singletonList(spec.get(0)
                    .withInitSqlScript("INSERT INTO EXISTING_DATA_ENTITYX (ID, NAME) VALUES (42, 'Foo')"));
        return spec;
    }
}

File: core-tests/e2e-tests/spring/spring-graphql/src/test/kotlin/com/foo/graphql/db/exisitingdata/ExistingDataController.kt
package com.foo.graphql.db.exisitingdata

import com.foo.graphql.db.SpringWithDbController
import org.evomaster.client.java.sql.DbSpecification


class ExistingDataController : SpringWithDbController(ExistingDataApplication::class.java) {

    override fun schemaName() = "dbexisting.graphqls"


    override fun resetStateOfSUT() {
        super.resetStateOfSUT()

//        val rep = ctx!!.getBean<ExistingDataXRepository>(ExistingDataXRepository::class.java)
//        rep.save(ExistingDataX(42L, "Foo"))
    }

    override fun getDbSpecifications(): MutableList<DbSpecification>? {
        val spec = super.getDbSpecifications()
        if (spec != null && spec.isNotEmpty())
            return mutableListOf(spec[0].withInitSqlScript("INSERT INTO EXISTING_DATAX (ID, NAME) VALUES (42, 'Foo')"))
        return spec
    }
}